pkg/services/pluginsintegration/config/envvars.go

```diff
@@ -0,0 +1,206 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/util"
+
+	"github.com/grafana/grafana-aws-sdk/pkg/awsds"
+	"github.com/grafana/grafana-azure-sdk-go/azsettings"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/proxy"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/config"
+	"github.com/grafana/grafana/pkg/plugins/envvars"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+var _ envvars.Provider = (*PluginEnvVarsProvider)(nil)
+
+type PluginEnvVarsProvider struct {
+	cfg             *setting.Cfg
```

Comment:
continuing my message above, I think it's easier to reason that whatever is within `pluginsintegration` only depends on the scoped plugin config so we don't need to handle two different configs here.